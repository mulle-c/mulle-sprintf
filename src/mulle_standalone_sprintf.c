//
//  mulle_standalone_sprintf.c
//  mulle-sprintf
//
//  Created by Nat! on 08.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#include "mulle_standalone_sprintf.h"

#include <stdint.h>


// food for the linker
uint32_t mulle_standalone_sprintf_version = MULLE_SPRINTF_VERSION;
