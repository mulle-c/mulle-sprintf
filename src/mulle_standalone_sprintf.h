//
//  mulle_standalone_sprintf.h
//  mulle-sprintf
//
//  Created by Nat! on 08.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//

#ifndef mulle_standalone_sprintf_h__
#define mulle_standalone_sprintf_h__

#include <mulle_sprintf/mulle_sprintf.h>

#endif 
