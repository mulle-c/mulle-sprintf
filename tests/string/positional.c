#include <mulle_standalone_sprintf/mulle_standalone_sprintf.h>
#include <mulle_test_allocator/mulle_test_allocator.h>
#include <mulle_container/mulle_container.h>

#include <stdio.h>



static void   complex_width_tests()
{
   struct mulle_buffer   *buffer;

   buffer = mulle_buffer_create( NULL);

   mulle_sprintf( buffer, "%2$*1$s/%.*s\n", 3, "VfL Bochum 1848", 5, "VfL Bochum 1848");
   printf( "%s", mulle_buffer_get_bytes( buffer));

   mulle_sprintf( buffer, "%*.*s/%*.*s\n", 3, 5, "VfL Bochum 1848", 5, 3, "VfL Bochum 1848");
   printf( "%s", mulle_buffer_get_bytes( buffer));
   mulle_buffer_reset( buffer);

   mulle_sprintf( buffer, "%*s/%.*s\n", 3, "VfL Bochum 1848", 5, "VfL Bochum 1848");
   printf( "%s", mulle_buffer_get_bytes( buffer));

   mulle_buffer_destroy( buffer);
}

int  main()
{
   mulle_test_allocator_set_tracelevel( 1);
   mulle_test_allocator_initialize();
   mulle_default_allocator = mulle_test_allocator;

   complex_width_tests();

   mulle_sprintf_free_storage();
   mulle_test_allocator_reset();
}

