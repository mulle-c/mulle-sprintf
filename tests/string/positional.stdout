VfL Bochum 1848/VfL B
VfL Bochum 1848/VfL B
VfL Bochum 1848/VfL B
